### PR TITLE
Sensor Config is introduced

### PR DESCRIPTION
**Pull request by shybzzz**

### CODE REVIEW

File: init-sensor.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#ifndef __INIT_SENSOR__
+#define __INIT_SENSOR__
+
+using getData = int();
+
+struct SensorConfig {
```
Comment (ihor96berizka) [inline init-sensor.h:6]: Will this struct be extended?
If no, there is no need to define new type using struct. We can use uint8_t. Or, we can use typedef\using statement to define new type(same as uint8_t, but different name).

File: init-sensor.h [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  initSensorType();
+}
+
+int gatherData(getData func) {
+  int d = func();
+  if (current_sample < DATA_SIZE) {
+    data[current_sample] = d;
+    current_sample++;
+  } else {
+    for (int lc = 1; lc < DATA_SIZE; lc++) {
```
Comment (ihor96berizka) [inline init-sensor.h:62]: This loop was rewritten using std::copy(), wasn`t it?

File: init-server.h [hunk 3]
```diff
@@ -18,47 +16,108 @@ void printSocketBuff() {
 bool readSocket(WiFiClient& client, uint8_t dataType) {
   bool res = false;
   uint16_t msgLength = client.available();
-  
-  if(msgLength > 0 && dataType == requestHeader) {
+
+  if (msgLength > 0 && dataType == requestHeader) {
     memset(socketBuff, '\0', MAX_SOCKET_BUFF_SIZE);
     char* tmp = nullptr;
```
Comment (ihor96berizka) [inline init-server.h:22]: I think that would be nice to define this variable on stack - char tmp[msgLen]. Try to replace old definition with this one(May not work, but at least we need to try). To test this approach - remove also lines 22 and 27-28

File: init-sensor.h (hunk 1 above)
Comment (shybzzz) [reply init-sensor.h:6]: Yes this is not going to be single property
We'll have some more configurations

File: init-sensor.h (hunk 2 above)
Comment (shybzzz) [reply init-sensor.h:62]: Somewhere in another place
Will check that

File: init-server.h (hunk 3 above)
Comment (shybzzz) [reply init-server.h:22]: We need much more refactoring here and in another place
Let us meet to discuss that and let it be a part of another pull request

**Pull request merged**